packages/react/src/components/Select/index.test.tsx

```diff
@@ -0,0 +1,263 @@
+import { renderToString } from 'react-dom/server'; // convert the React element to HTML
+import Select from './';
+import axe from '../../axe';
+
+const defaultProps = {
+  name: 'Test Select',
+  label: 'Test Select'
+};
+
+const withCustomOptions = (otherProps = {}) => {
+  const options = [
+    { key: '1', value: 'Fred' },
+    { key: '2', value: 'Bill' },
+    { key: '3', value: 'Ted' },
+    { key: '4', value: 'Bob' }
+  ];
+
+  return render(
+    <div>
+      <Select
+        {...defaultProps}
+        {...otherProps}
+        defaultValue="Bill"
+        options={options}
+      />
+    </div>
+  );
+};
+
+test('Should render the expected UI elements', () => {
+  withCustomOptions();
+
+  expect(screen.getByText(defaultProps.name)).toBeInTheDocument();
+  expect(
+    screen.getByText(defaultProps.name).parentElement?.parentElement
+  ).toHaveClass('Field__select');
+  expect(screen.getAllByRole('option')[0]).toHaveClass('Field__option');
+  expect(screen.getAllByRole('option')).toHaveLength(4);
+  expect(screen.queryByText('Required')).toBeNull();
+  expect(screen.queryByText('Error')).toBeNull();
+});
+
+test('Should set option attributes properly', () => {
+  render(
+    <Select
+      {...defaultProps}
+      defaultValue="a"
+      options={[
+        { key: '1', value: 'a' },
+        { disabled: true, key: '2', value: 'b' },
+        { key: '3', value: 'c' }
+      ]}
+    />
+  );
+
+  expect(screen.getAllByRole('option')).toHaveLength(3);
+  expect(screen.getByText('b')).toBeInTheDocument();
+});
+
+test('Should pass ref properly', () => {
+  const ref = React.createRef<HTMLSelectElement>();
+  render(
+    <Select
+      {...defaultProps}
+      id="test-id"
+      ref={ref}
+      defaultValue="a"
+      options={[{ key: '1', value: 'a' }]}
+    />
+  );
+
+  expect(ref.current).toBeInTheDocument();
+  expect(ref.current).toHaveAttribute('id', 'test-id');
+});
+
+test('Should pass children properly', () => {
+  render(
+    <Select {...defaultProps} defaultValue="a">
+      <option>a</option>
+      <option disabled>b</option>
+      <option>c</option>
+    </Select>
+  );
+
+  expect(screen.getAllByRole('option')).toHaveLength(3);
+  expect(screen.getByText('b')).toBeInTheDocument();
+});
+
+test('Should render required text', () => {
+  withCustomOptions({ required: true });
+  expect(screen.getByText('Required')).toBeInTheDocument();
+
+  withCustomOptions({ required: true, requiredText: 'Bananas' });
+  expect(screen.getByText('Bananas')).toBeInTheDocument();
+});
+
+test('Should handle errors', () => {
+  const errorText = 'ErR0r';
+  withCustomOptions({ error: errorText });
+
+  expect(screen.getByText(errorText)).toBeInTheDocument();
+  expect(screen.getByText(errorText)).toHaveAttribute(
+    'id',
+    screen.getByText(errorText).id
+  );
+  expect(screen.getByRole('combobox')).toHaveAttribute(
+    'aria-describedby',
+    screen.getByText(errorText).id
+  );
```

Comment:
In this instance, we can use 
```suggestion
  expect(screen.getByRole('combobox')).toHaveAccessibleDescription(errorText);
```